### ISSUE TITLE
biblatex >= 3.12 does not honour date-range format for cross-referenced entries

### ISSUE THREAD

**Issue by sicherha** [plk/biblatex#913]
I have a bibliography entry (for a conference paper) that cross-references another entry (for the corresponding conference) which contains a date range. When processing that first entry, biblatex fails to honour the format specification for the date range.

## Example
Consider the following minimal example - note the `date=comp` option:
```latex
\documentclass{article}

\usepackage[date=comp]{biblatex}
\usepackage{filecontents}

\begin{filecontents}{daterange-bug.bib}
	@inproceedings{bibentry,
		author    = {Author},
		title     = {Title},
		crossref  = {crossreference},
	}
	@proceedings{crossreference,
		booktitle = {Book Title},
		date      = {2019-08-19/2019-08-21},
	}
\end{filecontents}

\addbibresource{daterange-bug.bib}

\begin{document}
	\nocite{bibentry}
	\printbibliography
\end{document}
```
### Expected output
I expect the date range to be in the compact (`comp`) format:
> [1] Author. “Title”. In: Book Title. Aug. 19–21, 2019.

### Actual output
What I actually get is the long format:
> [1] Author. “Title”. In: Book Title. Aug. 19, 2019–Aug. 21, 2019.

## Narrowing it down
The issue does not occur if the two entries are merged into one.
I can reproduce the bug with biblatex 3.12 and the current `master`; biblatex 3.11 works fine.

According to `git bisect`, the faulty behaviour was introduced with commit 9a1f550222e294af5490488093a5e33e10493c63.

**Comment by moewew** [comment]
Thank you for reporting this bug. And thank you also for finding out where the culprit is hidden.

@plk This seems to be a Biber issue. `dateera` and `enddateera` are not present in the `.bbl` file, it seems they are not copied when the `date` is copied over from `crossref` (similar to https://github.com/plk/biblatex/issues/897).

If we can't get a fix for Biber into 3.13 because building the binaries has already started, I could offer a temporary fix for this particular issue in `biblatex`. Let me know if you want me to commit it to any of the two `biblatex` branches
```latex
\def\blx@imc@ifdateyearsequal#1#2{%
  \ifboolexpr{ test {\iffieldsequal{#1year}{#2year}}
               and
               (test {\iffieldsequal{#1dateera}{#2dateera}}
                or
                (    test {\iffieldundef{#1dateera}}
                 and test {\iffieldundef{#2dateera}}))}}
```

**Comment by sicherha** [comment]
Thank you for the very quick response! I can confirm that the workaround fixes the issue for me.

**Comment by plk** [comment]
Should be fixed in biber v2.13 which is being rebuilt and uploaded now. It's fairly safe as we haven't submitted to CTAN yet.

**Comment by moewew** [comment]
Thank you for the quick fix. That resolved the problem in my tests. I see that all Biber builds that were built already are re-uploaded now, so we should be good.

**Comment by moewew** [comment]
Fixed in `biblatex` 3.13/Biber 2.13 released yesterday.
